src/meta/src/stream/scheduler.rs

```diff
@@ -12,84 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::{BTreeMap, HashMap, LinkedList};
-use std::iter::empty;
+use std::collections::{BTreeMap, HashMap};
 
-use anyhow::{anyhow, Context};
 use itertools::Itertools;
-use rand::prelude::SliceRandom;
-use risingwave_common::bail;
-use risingwave_common::buffer::Bitmap;
-use risingwave_common::hash::ParallelUnitMapping;
-use risingwave_pb::common::{ActorInfo, Buffer, ParallelUnit, WorkerNode};
-use risingwave_pb::meta::table_fragments::fragment::FragmentDistributionType;
-use risingwave_pb::meta::table_fragments::Fragment;
+use risingwave_pb::common::{ActorInfo, ParallelUnit};
 
 use crate::manager::{WorkerId, WorkerLocations};
 use crate::model::ActorId;
-use crate::MetaResult;
 
-/// [`Scheduler`] defines schedule logic for mv actors.
-pub struct Scheduler {
-    /// The parallel units of the cluster in a round-robin manner on each worker.
-    all_parallel_units: Vec<ParallelUnit>,
-}
-
-/// [`ScheduledLocations`] represents the location of scheduled result.
-pub struct ScheduledLocations {
+/// [`Locations`] represents the parallel unit and worker locations of the actors.
+#[cfg_attr(test, derive(Default))]
+pub struct Locations {
     /// actor location map.
     pub actor_locations: BTreeMap<ActorId, ParallelUnit>,
     /// worker location map.
     pub worker_locations: WorkerLocations,
-    /// actor vnode bitmap.
-    pub actor_vnode_bitmaps: HashMap<ActorId, Option<Buffer>>,
```

Comment:
The bitmap is already filled when building the actor graph, so we don't need this anymore. This structure is now only used for deciding which compute node the requests should be dispatched to, i.e., the RPC part of the `create_job_impl`.